systemd-boot/default-settings

```diff
@@ -10,31 +10,39 @@
 # include common functions
 . "$PBL_INCLUDE/library"
 
-target=$(uname -m)
-
-case "$target" in
-  x86_64|i?86)
-    image=vmlinuz ;;
-  ppc*)
-    image=vmlinux ;;
-  s390*)
-    image=image ;;
-  armv*)
-    image=zImage ;;
-  aarch64|riscv64)
-    image=Image ;;
-  *)
-    echo "Architecture $target not supported."
-esac
-
 lib_get_cmdline_bls
-
-kernel=$(readlink -f "/boot/$image")
-initrd=$(readlink -f /boot/initrd)
 append=\"$(lib_shellquote "$cmdline")\"
 
+err=0
+if [ -x /usr/bin/bootctl ] ; then
+  kernel=$(bootctl status | grep linux: | cut -d ':' -f 2 | tr -ds ' ' '/')
+  initrd=$(bootctl status | grep initrd: | cut -d ':' -f 2 | tr -ds ' ' '/')
+else
```

Comment:
This fallback is dangerous. The reasons are two:

* There is no component that synchronize the links when a default entry is selected (not bootctl, suse-module-tools, not sdbootutil)
* If the image has links and is using BLS, then there is a bug in suse-module-tools, and the links will be broken